File: tools-ui/src/app/model/spending-transaction.ts
export class SpendingTransaction {
  id: number | undefined
  accountId: number | undefined
  identifier: string | undefined
  name: string | undefined
  originalName: string | undefined
  amount: string | undefined
  category: string | undefined
  location: string | undefined
  date: string | undefined

  error: boolean | undefined

  public constructor(init?:Partial<SpendingTransaction>) {
    Object.assign(this, init)
  }
}

File: tools-server/src/main/java/com/fmning/tools/domain/SpendingTransaction.java
package com.fmning.tools.domain;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.DynamicUpdate;

import java.sql.Date;

@Entity
@Table(name="tl_spending_transactions")
@DynamicUpdate
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class SpendingTransaction {

    @Id
    @Column(name="id")
    @GeneratedValue(strategy= GenerationType.IDENTITY)
    private int id;

    @Column(name="account_id")
    private int accountId;

    @Column(name="identifier")
    private String identifier;

    @Column(name="name")
    private String name;

    @Column(name="original_name")
    private String originalName;

    @Column(name="amount")
    private String amount;

    @Column(name="category")
    private String category;

    @Column(name="location")
    private String location;

    @Column(name="date")
    private Date date;

}
